Patcher/BinarySearcher.cs

```diff
@@ -0,0 +1,99 @@
+namespace Patcher
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Original by Matthew Watson from https://stackoverflow.com/questions/37500629/find-byte-sequence-within-a-byte-array
+    /// </summary>
+    public sealed class BinarySearcher
```

Comment:
Did you benchmark this against the current three line implementation? I did have something similar to this drafted up initially, but appeal of having just three lines outweighed the performance benefits of what I wrote at the time. Would be interesting to see if this is hugely faster than the current imeplementation.

Reply:
I haven't run the numbers on it yet. It finds the Hex pretty much instantly instead of waiting a couple of seconds, so it at least feels(TM) a lot faster. I can benchmark it, probably tomorrow, to give some hard numbers and we can discuss if it is worth the, admittedly, more complex code.

Reply:
I ran the numbers on the simple three line loop and the BinarySearcher, each had 100 runs and was looking for a Pattern it found and one it didn't find. I ran it against Unity 2019.3.9f1 (165mb size)
Loop: Total: 5501.685s, Min: 51.490s, Max: 61.231s, Avg: 55.016s
BinarySearcher: Total: 69.966s, Min: 609ms, Max: 864ms, Avg: 699.66ms

I think its worth the ~ 80x speed increase, but its your call if we want to have the more complex Code to maintain.

P.S. If my testing methodology was flawed let me know and I can rerun the benchmark.

Reply:
Sounds good! I don't think pattern matching code requires a whole lot of maintenance, so the performance benefits definitely outweigh the maintenance overhead in my opinion.